src/Hl7.Fhir.Support/Serialization/PrimitiveTypeConverter.cs

```diff
@@ -120,7 +121,14 @@ private static object convertXmlStringToPrimitive(Type to, string value)
             if (typeof(DateTime) == to)
                 return ConvertToDatetimeOffset(value).UtcDateTime;  // Obsolete: use DateTimeOffset instead!!
             if (typeof(Decimal) == to)
-                return XmlConvert.ToDecimal(value);
+            {
+                if (Array.Exists(new[] { "+", ".", "00" }, c => value.StartsWith(c)) || value.EndsWith("."))
```

Comment:
This results in constructing this array for every single parse.  I think we could define:

```
    static readonly string[] FORBIDDEN_PREFIXES = new[] {...};
```

to avoid that.

Reply:
And I notice that if the array has 'N' values, `value.Endswith(.)` is also executed 'N' times.

Reply:
Alternatief: `if( Array.Exists(new[] { "+", ".", "00" }, c => value.StartsWith(c))) ||  value.EndsWith("."))`

Reply:
And you could replace the Exists with the expression `FORBIDDEN_PREFIXES.Any(c => value.StartsWith(c))`

Reply:
> And I notice that if the array has 'N' values, `value.Endswith(.)` is also executed 'N' times.

That is not really true. The if-statement is constructed by two operands: 
- Array.Exists() 
- value.EndsWith() 
The operator is an OR between these operands.

So first the Array.Exists() will be evaluated. If that results in false, then the second operand will be evaluated.

Reply:
> And you could replace the Exists with the expression `FORBIDDEN_PREFIXES.Any(c => value.StartsWith(c))`

I will change this